### DISCUSSION TITLE
Any way to install Gleam without admin permissions on Windows?

### DISCUSSION THREAD

**Discussion by mybearworld** [gleam-lang/gleam#2588] [Questions & support]
I was able to install Gleam, but it's not really usable:

```
  Resolving versions
Downloading packages
 Downloaded 2 packages in 0.17s
  Compiling gleam_stdlib
error: Program not found

The program `escript` was not found. Is it installed?
Documentation for installing Erlang can be viewed here:
https://gleam.run/getting-started/installing/
```

Going to that page, both chocolatey and [this](http://erlang.org/download/otp_win64_26.2.2.exe) seem to require admin privileges, which I don't have.

Is there any way to do this without admin privileges?

**Comment by mybearworld** [comment]
I have found https://stackoverflow.com/q/2688873/, but the answers there aren't very specific, and I'm not sure what to do.

**Comment by mybearworld** [comment]
Oh, I can just set the target to JS. Pretty annoying because it looks like Erlang is the recommended and better choice, though. But this is out of scope for Gleam, then.

**Comment by lpil** [comment]
Hello! Escript is part of Erlang. I'm afraid I am unable to provide advice about installing Erlang on Windows as I don't maintain or know about any tools or approaches for this. I hope you are able to find a good solution, sorry I cannot help more.

**Comment by mardukbp** [comment]
I also don't have admin privileges and wanted to install Elixir. I successfully [modified the LiveBook installer](https://nc.nl.tab.digital/s/K2Qspe8epe9a2Cm) so that it does not require admin privileges and got Elixir working :) 

After the installation you have to adjust the file `%LOCALAPPDATA%\Livebook\rel\vendor\otp\bin\erl.ini`:

```
[erlang]
Bindir=C:\\Users\\YourUser\\AppData\\Local\\Livebook\\rel\\vendor\\otp\\erts-14.2.5\\bin
Progname=erl
Rootdir=C:\\Users\\YourUser\\AppData\\Local\\Livebook\\rel\\vendor\\otp
```

And you have to add the following directories to the %PATH% environment variable:

```
C:\Users\YourUser\AppData\Local\Livebook\rel\vendor\otp\bin
C:\Users\YourUser\AppData\Local\Livebook\rel\vendor\elixir\bin
```

Then you should be able to use Erlang, Elixir and Gleam :)
